src/showingpreviously/cinemas/omniplex.py

```diff
@@ -0,0 +1,97 @@
+from datetime import datetime, timedelta
+from bs4 import BeautifulSoup
+import re
+
+import showingpreviously.requests as requests
+from showingpreviously.model import ChainArchiver, CinemaArchiverException, Chain, Cinema, Screen, Film, Showing
+from showingpreviously.consts import STANDARD_DAYS_AHEAD, UK_TIMEZONE, UNKNOWN_FILM_YEAR, UNKNOWN_SCREEN
+
+
+CINEMAS_URL = 'https://www.omniplex.ie/'
+SHOWINGS_URL = 'https://www.omniplex.ie/cinema/{cinema_id}'
+SCREEN_NAME_PATTERN = re.compile(r'var\s+screen\s*=\s*\'(?P<screen_name>.+?)\';')
+
+CHAIN = Chain('Omniplex Cinemas')
+
+
+def get_response(url: str) -> requests.Response:
+    r = requests.get(url)
+    if r.status_code != 200:
+        raise CinemaArchiverException(f'Got status code {r.status_code} when fetching URL {url}')
+    return r
+
+
+def get_cinemas_as_dict() -> dict[str, Cinema]:
+    r = get_response(CINEMAS_URL)
+    soup = BeautifulSoup(r.text, features='html.parser')
+    cinema_select = soup.find('select', {'id': 'homeSelectCinema'})
+    cinemas = {}
+    for cinema_option in cinema_select.find_all('option', {'id': True, 'data-thissite': True}):
+        cinema_name = cinema_option.text.strip()
+        cinema_id = cinema_option['id']
+        cinemas[cinema_id] = Cinema(cinema_name, UK_TIMEZONE)
+    return cinemas
+
+
+def get_attributes(at: str) -> dict[str, any]:
+    at = at.lower()
+    attributes = {'format': []}
+    if 'subtitled' in at:
+        attributes['subtitled'] = True
+    if 'sensory' in at:
+        attributes['format'].append('sensory')
+    if len(attributes['format']) == 0:
+        del attributes['format']
+    return attributes
+
+
+def get_showing_screen(booking_link: str) -> Screen:
+    r = get_response(booking_link)
+    if 'Booking Error' in r.text:
+        return UNKNOWN_SCREEN
```

Comment:
I am slightly concerned that if the API / site changes, then it may end up failing silently, filling in all the screens as unknown when they are known (just that the script needs adjusting).

Reply:
Once the new error system is added, we can log warnings here.